src/hotspot/share/opto/opaquenode.hpp

```diff
@@ -132,6 +132,20 @@ class Opaque4Node : public Node {
   virtual const Type* Value(PhaseGVN* phase) const;
 };
 
+// This node is used for Initialized Assertion Predicate BoolNodes. Initialized Assertion Predicates must always evaluate
+// to true. Therefore, we get rid of them in product builds as they are useless. In debug builds we keep them as
+// additional verification code (i.e. removing this node and use the BoolNode input instead).
+class OpaqueInitializedAssertionPredicateNode : public Node {
```

Comment:
Shouldn't the new OpaqueInitializedAssertionPredicateNode be a subclass of Opaque4 or shouldn't both be a subclass of a common super type? Don't they share at least some logic or behavior?

Reply:
Fair enough.